src/session/sidebar/avatar.rs

```diff
@@ -204,17 +204,15 @@ impl Avatar {
             binding.unwatch();
         }
 
+        self.imp().avatar.set_item(item.clone());
+
         if let Some(ref item) = item {
             if let Some(chat) = item.downcast_ref::<Chat>() {
-                imp.avatar.set_item(Some(chat.avatar().to_owned()));
-
                 match chat.type_().user() {
                     Some(user) => self.setup_is_online_binding(user),
                     None => self.set_is_online(false),
                 }
             } else if let Some(user) = item.downcast_ref::<User>() {
-                imp.avatar.set_item(Some(user.avatar().to_owned()));
-
```

Comment:
Mh, why these are unrelated? The imp.avatar now requires just the User or Chat objects instead of the AvatarItem like it was before, so it makes sense to just pass the item object directly.

Reply:
Ah, correct. I did not realize that. Shame on me. :-P